File: src/fabric/common/main/java/settingdust/mod_sets/fabric/ModSetsModMenu.java
package settingdust.mod_sets.fabric;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import settingdust.mod_sets.game.ModSetsConfigScreenGenerator;

public class ModSetsModMenu implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return ModSetsConfigScreenGenerator::generateScreen;
    }
}

File: src/fabric/common/main/kotlin/settingdust/mod_sets/fabric/ModSetsModMenu.kt
package settingdust.mod_sets.fabric

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import settingdust.mod_sets.game.ModSetsIngameConfig

class ModSetsModMenu : ModMenuApi {
    override fun getModConfigScreenFactory() = ConfigScreenFactory {
        ModSetsIngameConfig.generateConfigScreen(it)
    }
}
